push/src/curl_wrapper.h

```diff
@@ -21,11 +22,13 @@ class CurlWrapper {
 
   int performHttpRequest(HttpMethod method, const std::string& uri,
                          const std::string& body);
+  void addOptHttpHeader(const std::string& header);
 
  private:
   CURL* curl_;
   std::string auth_;
   std::mutex mutex_;
+  std::vector<std::string> optHttpHeader_;
```

Comment:
I'd pre-compute and store the `curl_slist` here instead of a vector. Usually you add headers only once or twice. The sending happens much more often.